assets/src/components/v2/departures/departure_time.tsx

```diff
@@ -37,7 +37,7 @@ interface TimestampDeparture {
   minute: number;
 }
 
-const DepartureTime = ({ type, ...data }: Props) => {
+const DepartureTime: ComponentType<DepartureTime> = ({ type, ...data }) => {
```

Comment:
I used this pattern in places where a distinct "object" returned by the server corresponds exactly to the props of the component which renders an instance of that object. A `DepartureTime` (value) is a component which renders a `DepartureTime` (type). Within their namespaces, I think it makes sense to call each one of these by that name. But things don't always line up this way, as with e.g. the component one level above this, `DepartureTimes`: in this case the "props type" has a distinct name and is not exported.

```ts
export type TimeWithCrowding = { /* ... */ };

type Props = { timesWithCrowding: TimeWithCrowding[] };

const DepartureTimes: ComponentType<Props> = ({ timesWithCrowding }) => { /* ... */ };
```

I'd hesitate to say there's an "established pattern", since unfortunately a lot of the existing codebase just doesn't have explicit types at all (which is why we currently allow implicit-any). If anything it currently leans towards using separately-named "props types" in all cases. So I guess I'm proposing that we establish this as a pattern 🙂 What do you think?

In any case, thanks for calling this out, since I realized we should be using `import type` in places where we only use the type. It looks like the linter [has a rule to enforce this](https://typescript-eslint.io/rules/consistent-type-imports) but it's off by default. I'll see what happens when we turn it on.

Reply:
Okay, the lint rule gives 162 errors on the current codebase... it looks like it's less sophisticated than I thought, and just errors _whenever_ you import a type without using `type`, even if the name has no overlap with a value. (Though it seems like this could have advantages for bundle size anyway, so maybe we should think about turning it on, separately?)

I looked at all the imports across changed files in this PR, but as far as I can see, there actually isn't any place where I imported a value+type name and only used the type, so I haven't added any `type` keywords here.

Reply:
> If anything it currently leans towards using separately-named "props types" in all cases. So I guess I'm proposing that we establish this as a pattern 🙂 What do you think?

This is my personal preference so I'm on-board!

As an aside, I also prefer to use something like [zod](https://zod.dev/) to parse external data to ensure that types for it are accurate and not just wishful thinking. 📌 We can put a pin in that for now though.